pkg/generator_test.go

```diff
@@ -2346,6 +2346,190 @@ func NewRequesterGenerics[TAny interface{}, TComparable comparable, TSigned cons
 	s.checkGeneration("generic.go", "RequesterGenerics", false, "", expected)
 }
 
+func (s *GeneratorSuite) TestGenericExpecterGenerator() {
+	expected := `// RequesterGenerics is an autogenerated mock type for the RequesterGenerics type
+type RequesterGenerics[TAny interface{}, TComparable comparable, TSigned constraints.Signed, TIntf test.GetInt, TExternalIntf io.Writer, TGenIntf test.GetGeneric[TSigned], TInlineType interface{ ~int | ~uint }, TInlineTypeGeneric interface {
+	~int | test.GenericType[int, test.GetInt]
+	comparable
+}] struct {
+	mock.Mock
+}
+
+type RequesterGenerics_Expecter[TAny interface{}, TComparable comparable, TSigned constraints.Signed, TIntf test.GetInt, TExternalIntf io.Writer, TGenIntf test.GetGeneric[TSigned], TInlineType interface{ ~int | ~uint }, TInlineTypeGeneric interface {
+	~int | test.GenericType[int, test.GetInt]
+	comparable
+}] struct {
+	mock *mock.Mock
+}
+
+func (_m *RequesterGenerics[TAny, TComparable, TSigned, TIntf, TExternalIntf, TGenIntf, TInlineType, TInlineTypeGeneric]) EXPECT() *RequesterGenerics_Expecter[TAny, TComparable, TSigned, TIntf, TExternalIntf, TGenIntf, TInlineType, TInlineTypeGeneric] {
+	return &RequesterGenerics_Expecter[TAny, TComparable, TSigned, TIntf, TExternalIntf, TGenIntf, TInlineType, TInlineTypeGeneric]{mock: &_m.Mock}
+}
+
+// GenericAnonymousStructs provides a mock function with given fields: _a0
+func (_m *RequesterGenerics[TAny, TComparable, TSigned, TIntf, TExternalIntf, TGenIntf, TInlineType, TInlineTypeGeneric]) GenericAnonymousStructs(_a0 struct{ Type1 TExternalIntf }) struct {
+	Type2 test.GenericType[string, test.EmbeddedGet[int]]
+} {
+	ret := _m.Called(_a0)
+
+	var r0 struct {
+		Type2 test.GenericType[string, test.EmbeddedGet[int]]
+	}
+	if rf, ok := ret.Get(0).(func(struct{ Type1 TExternalIntf }) struct {
+		Type2 test.GenericType[string, test.EmbeddedGet[int]]
+	}); ok {
+		r0 = rf(_a0)
+	} else {
+		r0 = ret.Get(0).(struct {
+			Type2 test.GenericType[string, test.EmbeddedGet[int]]
+		})
+	}
+
+	return r0
+}
+
+// RequesterGenerics_GenericAnonymousStructs_Call is a *mock.Call that shadows Run/Return methods with type explicit version for method 'GenericAnonymousStructs'
+type RequesterGenerics_GenericAnonymousStructs_Call[TAny interface{}, TComparable comparable, TSigned constraints.Signed, TIntf test.GetInt, TExternalIntf io.Writer, TGenIntf test.GetGeneric[TSigned], TInlineType interface{ ~int | ~uint }, TInlineTypeGeneric interface {
+	~int | test.GenericType[int, test.GetInt]
+	comparable
+}] struct {
+	*mock.Call
+}
+
+// GenericAnonymousStructs is a helper method to define mock.On call
+//  - _a0 struct{Type1 TExternalIntf}
+func (_e *RequesterGenerics_Expecter[TAny, TComparable, TSigned, TIntf, TExternalIntf, TGenIntf, TInlineType, TInlineTypeGeneric]) GenericAnonymousStructs(_a0 interface{}) *RequesterGenerics_GenericAnonymousStructs_Call[TAny, TComparable, TSigned, TIntf, TExternalIntf, TGenIntf, TInlineType, TInlineTypeGeneric] {
```

Comment:
```
GenericAnonymousStructs(_a0 interface{})
```

Can you explain this bit? Should `_a0` have a type of `struct{Type1 TExternalIntf}`?